`dbt_cloud_api_token` in warehouse yaml breaks `soda scan`

**Describe the bug**
when the `dbt_cloud_api_token` key is added into the `warehouse.yaml` file, `soda scan` fails because the parsing of the warehouse yaml encounters a key that is not valid.

We probably want to make sure that this key is ok with `soda scan` when the dbt package is installed  or just fine full stop

version: latest main

```
  [warning] (dbt_cloud_api_token) Invalid key in telemetry/warehouse.yml : dbt_cloud_api_token
  | If you think this is a bug in Soda SQL, please open an issue at: https://github.com/sodadata/soda-sql/issues/new/choose
  | Exiting with code 1
  | Starting new HTTPS connection (1): collect.soda.io:443
  | https://collect.soda.io:443 "POST /v1/traces HTTP/1.1" 200 0

1   | 2.1.2
  | Scanning /Users/bastienboutonnet/repos/intelligence_analytics/soda_checks/telemetry/tables/soda_sql_datasource_sessions.yml ...
  | Injecting ssl_wrap_socket_with_ocsp
  | cache directory: /Users/bastienboutonnet/Library/Caches/Snowflake
  | Failed to import pyarrow. Cannot use pandas fetch API
  | Failed to import ArrowResult. No Apache Arrow result set format can be used. ImportError: No module named 'snowflake.connector.arrow_result'
  | Invalid key in telemetry/warehouse.yml : dbt_cloud_api_token
  | Scan failed: telemetry/warehouse.yml configuration errors:
  [warning] (dbt_cloud_api_token) Invalid key in telemetry/warehouse.yml : dbt_cloud_api_token
Traceback (most recent call last):
  File "/Users/bastienboutonnet/repos/soda-sql/core/sodasql/cli/cli.py", line 460, in scan
    scan = scan_builder.build(offline=offline)
  File "/Users/bastienboutonnet/repos/soda-sql/core/sodasql/scan/scan_builder.py", line 165, in build
    parser.assert_no_warnings_or_errors()
  File "/Users/bastienboutonnet/repos/soda-sql/core/sodasql/scan/parser.py", line 118, in assert_no_warnings_or_errors
    raise AssertionError(f'{self.description} configuration errors: \n  '
AssertionError: telemetry/warehouse.yml configuration errors:
  [warning] (dbt_cloud_api_token) Invalid key in telemetry/warehouse.yml : dbt_cloud_api_token
```

--- a/core/sodasql/scan/parser.py
+++ b/core/sodasql/scan/parser.py
@@ -33,6 +33,9 @@ class ParseLog:
     def __str__(self):
         return f'[{self.level}] ({self.field}) {self.message}'
 
+    def is_error(self):
+        return self.level == ERROR
+
     def is_error_or_warning(self):
         return self.level == ERROR or self.level == WARNING
 
@@ -107,6 +110,12 @@ class Parser:
         for invalid_key in [configured_key for configured_key in context_iterable if configured_key not in valid_keys]:
             self.warning(f'Invalid key in {self._get_context_description()} : {invalid_key}', invalid_key)
 
+    def has_errors(self):
+        for log in self.logs:
+            if log.is_error():
+                return True
+        return False
+
     def has_warnings_or_errors(self):
         for log in self.logs:
             if log.is_error_or_warning():
@@ -114,7 +123,7 @@ class Parser:
         return False
 
     def assert_no_warnings_or_errors(self):
-        if self.has_warnings_or_errors():
+        if self.has_errors():
             raise AssertionError(f'{self.description} configuration errors: \n  '
                                  + ('\n  '.join([str(log) for log in self.logs])))
 

